File: 1768-merge-strings-alternately/1768-merge-strings-alternately.go
func mergeAlternately(word1 string, word2 string) string {
     n1, n2 := len(word1), len(word2)
    result := []byte{}
    
    for i := 0; i < n1 || i < n2; i++ {
        if i < n1 {
            result = append(result, word1[i])
        }
        if i < n2 {
            result = append(result, word2[i])
        }
    }
    return string(result)
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cs
public class Solution {
    public string MergeAlternately(string word1, string word2) {
        int maxLength = word1.Length >= word2.Length ? word1.Length : word2.Length;
        
        Console.WriteLine($"Max Length: {maxLength}");
        string merged = null;
        for(int i = 0; i < maxLength; i++){
            if (word1.Length > i)
                merged += word1[i];
            if (word2.Length > i)            
                merged += word2[i];
            Console.WriteLine($"Merged: {merged}");

        }

        return merged;
    }
}